client/state/account-recovery/reducer.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * External dependencies
+ */
+
+/**
+ * Internal dependencies
+ */
+import {
+	ACCOUNT_RECOVERY_FETCH_SUCCESS,
+} from 'state/action-types';
+
+const reducer = ( state = {}, action ) => {
```

Comment:
How about we use the `createReducer` function here in state/utils

It'd change to something like

```js
export const reducer = createReducer( {}, {
    [ ACCOUNT_RECOVERY_FETCH_SUCCESS ]: ( { accountRecoverySettings } ) => {
        const {
            email,
            email_validated: emailValidated,
            phone,
            phone_validated: phoneValidated,
        } = accountRecoverySettings;
        return { email, emailValidated, phone, phoneValidated };
    },
} );
```